packages/dd-trace/test/plugins/agent.js

```diff
@@ -235,70 +306,32 @@ module.exports = {
 
   // Register handler to be executed each agent call, multiple times
   subscribe (handler) {
-    handlers.add({ handler })
+    traceHandlers.add({ handler })
   },
 
   // Remove a handler
   unsubscribe (handler) {
-    handlers.delete(handler)
+    traceHandlers.delete(handler)
   },
 
   /**
    * Register a callback with expectations to be run on every tracing payload sent to the agent.
-   * If the callback does not throw, the returned promise resolves. If it does,
-   * then the agent will wait for additional payloads up until the timeout
-   * (default 1000 ms) and if any of them succeed, the promise will resolve.
-   * Otherwise, it will reject.
-   *
-   * @param {(traces: Array<Array<object>>) => void} callback - A function that tests trace data as it's received.
-   * @param {Object} [options] - An options object
-   * @param {number} [options.timeoutMs=1000] - The timeout in ms.
-   * @param {boolean} [options.rejectFirst=false] - If true, reject the first time the callback throws.
-   * @returns {Promise<void>} A promise resolving if expectations are met
    */
   use (callback, options) {
-    const deferred = {}
-    const promise = new Promise((resolve, reject) => {
-      deferred.resolve = resolve
-      deferred.reject = reject
-    })
-
-    const timeoutMs = options && typeof options === 'object' && options.timeoutMs ? options.timeoutMs : 1000
-
-    const timeout = setTimeout(() => {
-      if (error) {
-        deferred.reject(error)
-      }
-    }, timeoutMs)
-
-    let error
-    const handlerPayload = { handler, spanResourceMatch: options && options.spanResourceMatch }
-
-    function handler () {
-      try {
-        callback.apply(null, arguments)
-        handlers.delete(handlerPayload)
-        clearTimeout(timeout)
-        deferred.resolve()
-      } catch (e) {
-        if (options && options.rejectFirst) {
-          clearTimeout(timeout)
-          deferred.reject(e)
-        } else {
-          error = error || e
-        }
-      }
-    }
-
-    handler.promise = promise
-    handlers.add(handlerPayload)
+    return runCallback(callback, options, traceHandlers)
+  },
 
-    return promise
+  /**
+   * Register a callback with expectations to be run on every stats payload sent to the agent.
+   */
+  expectStats (callback, options) {
```

Comment:
```suggestion
  expectPipelineStats (callback, options) {
```
Let's change this to avoid confusion with DogStatsD stats.